editor/src/components/canvas/grid-panel.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import { NO_OP } from '../../core/shared/utils'
+import { UtopiaTheme, colorTheme } from '../../uuiui'
+import { LeftPanelMinWidth } from '../editor/store/editor-state'
+import { LeftPaneComponent } from '../navigator/left-pane'
+import { CodeEditorPane, ResizableRightPane } from './design-panel-root'
+import { useGridPanelDragInfo, useGridPanelDropArea } from './grid-panels-dnd'
+import type { GridPanelData, LayoutUpdate, StoredPanel } from './grid-panels-state'
+import { GridPanelVerticalGapHalf, GridPanelsNumberOfRows } from './grid-panels-state'
+
+interface GridPanelProps {
+  onDrop: (itemToMove: StoredPanel, newPosition: LayoutUpdate) => void
+  canDrop: (itemToMove: StoredPanel, newPosition: LayoutUpdate) => boolean
+  pane: GridPanelData
+}
+
+export const GridPanel = React.memo<GridPanelProps>((props) => {
+  const { onDrop, canDrop } = props
+  const { panel, index, span, order } = props.pane
+
+  const { isDragActive, draggedPanel } = useGridPanelDragInfo()
+
+  const dropAboveElement: LayoutUpdate = React.useMemo(
+    () => ({
+      type: 'before-index',
+      indexInColumn: order,
+      columnIndex: index,
+    }),
+    [order, index],
+  )
+
+  const dropBelowElement: LayoutUpdate = React.useMemo(
+    () => ({
+      type: 'after-index',
+      indexInColumn: order,
+      columnIndex: index,
+    }),
+    [order, index],
+  )
+
+  const canDropAbove = draggedPanel != null && canDrop(draggedPanel, dropAboveElement)
+  const canDropBelow = draggedPanel != null && canDrop(draggedPanel, dropBelowElement)
+
+  const { drop: dropBefore, isOver: isOverBefore } = useGridPanelDropArea(
+    index,
+    order,
+    React.useCallback(
+      (itemToMove: StoredPanel) => onDrop(itemToMove, dropAboveElement),
+      [onDrop, dropAboveElement],
+    ),
+  )
+  const { drop: dropAfter, isOver: isOverAfter } = useGridPanelDropArea(
+    index,
+    order,
+    React.useCallback(
+      (itemToMove: StoredPanel) => {
+        onDrop(itemToMove, dropBelowElement)
+      },
+      [onDrop, dropBelowElement],
+    ),
+  )
+
+  const draggablePanelComponent = (() => {
+    switch (panel.name) {
+      case 'code-editor':
+        return (
+          <CodeEditorPane
+            panelData={props.pane.panel}
+            small={props.pane.span !== GridPanelsNumberOfRows} // it's only not small if it's full-height (span === number of rows)
```

Comment:
So perhaps a better prop name would be `fullHeight`

Reply:
the prop is talking in terms of VSCode – small means it will get a 0.7x scaling. in the grid context, we decide that small happens when it's not full height in the grid